allennlp/modules/seq2vec_encoders/bert_pooler.py

```diff
@@ -0,0 +1,50 @@
+from typing import Union
+
+from overrides import overrides
+
+import torch
+from pytorch_pretrained_bert import BertModel
+
+from allennlp.modules.seq2vec_encoders.seq2vec_encoder import Seq2VecEncoder
+
+@Seq2VecEncoder.register("bert_pooler")
+class BertPooler(Seq2VecEncoder):
+    """
+    The pooling layer at the end of the BERT model. If you want to use the pretrained
+    BERT model to build a classifier and you want to use the AllenNLP token-indexer ->
+    token-embedder -> seq2vec encoder setup, this is the Seq2VecEncoder to use.
+    (For example, if you want to experiment with other embedding / encoding combinations.)
+
+    If you just want to train a BERT classifier, it's simpler to just use the
+    ``BertForClassification`` model.
+
+    Parameters
+    ----------
+    pretrained_model: ``Union[str, BertModel]``
+        The pretrained BERT model to use. If this is a string,
+        we will call ``BertModel.from_pretrained(pretrained_model)``
+        and use that.
+    """
+    def __init__(self, pretrained_model: Union[str, BertModel]) -> None:
+        super().__init__()
+
+        # TODO(joelgrus): it's inefficient to load the model here and (presumably) also in the
```

Comment:
You could take the approach we use for spacy models, where we basically have a static dictionary of loaded models.  This could get a little hairy if you're fine tuning (lots of potential crazy bugs if you use it wrong), but if it's private and the only use for it is sharing the token embedder and the pooler, it might be ok, because we'll do the right thing.

Reply:
done